plugin/storage/es/spanstore/writer_test.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package spanstore
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/olivere/elastic"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger/model"
+	"github.com/uber/jaeger/pkg/es/mocks"
+	"github.com/uber/jaeger/pkg/testutils"
+	"github.com/uber/jaeger/storage/spanstore"
+)
+
+type spanWriterTest struct {
+	client    *mocks.Client
+	logger    *zap.Logger
+	logBuffer *testutils.Buffer
+	writer    *SpanWriter
+}
+
+func withSpanWriter(fn func(w *spanWriterTest)) {
+	client := &mocks.Client{}
+	logger, logBuffer := testutils.NewLogger()
+	w := &spanWriterTest{
+		client:    client,
+		logger:    logger,
+		logBuffer: logBuffer,
+		writer:    NewSpanWriter(client, logger),
+	}
+	fn(w)
+}
+
+func TestNewSpanWriter(t *testing.T) {
+	withSpanWriter(func(w *spanWriterTest) {
+		var writer spanstore.Writer = w.writer
+		assert.NotNil(t, writer)
+	})
+}
+
+func TestSpanWriter_WriteSpan(t *testing.T) {
```

Comment:
I'll leave the large test and comment it up so it provides an insight to how `WriteSpan` should generally behave, and I'll add separate tests for the individual functions